src/borg/testsuite/repository.py

```diff
@@ -12,42 +11,40 @@
 from ..helpers import IntegrityError
 from ..helpers import msgpack
 from ..locking import Lock, LockFailed
+from ..platformflags import is_win32
 from ..remote import RemoteRepository, InvalidRPCMethod, PathNotAllowed
 from ..repository import Repository, LoggedIO, MAGIC, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT
 from ..repoobj import RepoObj
 from .hashindex import H
 
 
-class SingularRepositoryMaker:
-    # For now, this class is not used. Just a placeholder until I am sure if/how to incorporate it into project.
-    def __init__(self, exclusive=True, create=True):
-        self.current_repo = None
-        self.path = os.path.join(tempfile.mkdtemp(), "repository")
-        self.exclusive = exclusive
-        self.create = create
-        self.exit_stack = ExitStack()
-
-    def __enter__(self):
-        if self.current_repo is not None:
-            raise RuntimeError("Cannot re-enter a disposed repository.")
-        self.current_repo = self.exit_stack.enter_context(Repository(self.path, self.exclusive, self.create))
-        return self.current_repo
-
-    def __exit__(self, exc_type, exc_value, traceback):
-        # Decide what needs to be done here.
-        # self.context_stack.undo()  # perhaps pass over args here
-        self.current_repo = None
-
-
 @pytest.fixture()
 def repository(tmpdir):
     repository_location = os.path.join(str(tmpdir), "repository")
     repository = Repository(repository_location, exclusive=True, create=True)
     yield repository
 
 
-def reopen(repository, exclusive=True, create=False):
-    return Repository(repository.path, exclusive=exclusive, create=create)
+@pytest.fixture()
+def remote_repository(tmpdir):
+    if is_win32:
+        pytest.skip("On Windows, we skip the remote tests.")
+    repo_location = Location("ssh://__testsuite__" + os.path.join(str(tmpdir), "repository"))
+    return RemoteRepository(repo_location, exclusive=True, create=True)
+
+
+def reopen(repository, exclusive: Optional[bool] = True, create=False):
```

Comment:
Maybe we could put some sort of check here to ensure the repository we are reopening has already been closed?

Reply:
Yes. Guess that was also one part of what ronny suggested.